browser/ui/webui/brave_rewards_page_ui.cc

```diff
@@ -1858,45 +1856,49 @@ void RewardsDOMHandler::OnGetMonthlyReport(
   data.SetIntKey("year", year);
 
   base::Value balance_report(base::Value::Type::DICTIONARY);
-  balance_report.SetDoubleKey("grant", report.balance.grants);
-  balance_report.SetDoubleKey("ads", report.balance.earning_from_ads);
-  balance_report.SetDoubleKey("contribute", report.balance.auto_contribute);
-  balance_report.SetDoubleKey("monthly", report.balance.recurring_donation);
-  balance_report.SetDoubleKey("tips", report.balance.one_time_donation);
+  balance_report.SetDoubleKey("grant", report->balance->grants);
+  balance_report.SetDoubleKey("ads", report->balance->earning_from_ads);
+  balance_report.SetDoubleKey("contribute", report->balance->auto_contribute);
+  balance_report.SetDoubleKey("monthly", report->balance->recurring_donation);
+  balance_report.SetDoubleKey("tips", report->balance->one_time_donation);
 
   base::Value transactions(base::Value::Type::LIST);
-  for (const auto& item : report.transactions) {
+  for (const auto& item : report->transactions) {
     base::Value transaction_report(base::Value::Type::DICTIONARY);
-    transaction_report.SetDoubleKey("amount", item.amount);
-    transaction_report.SetIntKey("type", item.type);
-    transaction_report.SetIntKey("processor", item.processor);
-    transaction_report.SetIntKey("created_at", item.created_at);
+    transaction_report.SetDoubleKey("amount", item->amount);
+    transaction_report.SetIntKey("type", static_cast<int>(item->type));
+    transaction_report.SetIntKey(
+        "processor",
+        static_cast<int>(item->processor));
+    transaction_report.SetIntKey("created_at", item->created_at);
 
     transactions.Append(std::move(transaction_report));
   }
 
   base::Value contributions(base::Value::Type::LIST);
-  for (const auto& item : report.contributions) {
+  for (const auto& item : report->contributions) {
     base::Value publishers(base::Value::Type::LIST);
-    for (const auto& item : item.publishers) {
+    for (const auto& item : item->publishers) {
```

Comment:
Might make sense to rename one of these variables to avoid confusion/mistakes in the future.

Reply:
fixed